readme.md

```diff
@@ -0,0 +1,176 @@
+# Tech Spec - Teleport Challenge
+
+## Overview
+To be implemented:
+
+1. `Job` library for starting, stopping, and getting status and streamed logs of jobs.
+2. API that implements the `Job` library, permitting clients to call the 4 public methods.
+3. CLI Client that calls API, using command line args to specify API methods.
+
+## Job Library
+`Job` type will include public fields:
+
+- Command
+- Args
+- ProcessID
+
+##### Implement methods for:
+
+- Start
+	- Assure job command exists. Executables can be located in Go using `exec.LookPath()`.
+	- Assure that a duplicate command is not currently running by checking "DB".
+	- Execute job w/ args.
+	- Write job stdout & stderr to specified dir.
+	- Set `ProcessID` on `Job`.
+- Stop
+	- Kill process.
+	- Delete job from "DB".
+- Get status
+	- Check "DB".  
+	- Return "running/complete" and logs.
+- Stream logs
+
+##### Tradeoffs
+- Killing the process, rather than sending SIGINT/SIGQUIT and waiting, comes with the risks associated with kill, e.g. lingering resources. However, it seems like the most user-intuitive option. Future versions could permit users to specify how they want to "stop" a job.
+- This implemenatation will prevent duplicate jobs from running simultaneously. There is conceivably some use in permitting redundant jobs, but it requires a more discussion around use cases (e.g. will this tool be used by customers in automation?)
+
+##### Limitations
+- For this milestone, mock DB with a global map rather than integrate with live database.
+
+## API
+Implement handlers for:
+
+- Start Job (unary)
+	- write logs to temp dir
+- Stop Job (unary)
+- Status (unary)
+- Stream Logs (stream)
+	- handles ongoing log output for long-running process
+
+Include interceptors for:
+
+- Authorization/Authentication
+	- stream & unary interceptors will authenticate user by examining the certificate-bound access token and authorize using role(s) from the access token.
+
+- Authentication
+	- a second layer of authentication will be provided via the client-end of mTLS. The API will verify the client cert with the CA.
+
+##### Tradeoffs
+- Writing logs to temp files means logs will be lost when temp dirs are purged, but will help prevent filling the hard disk with logs in the event of a runaway process.
+- Certificate-Bound Access Tokens are more complex than relying solely on mTLS for authentication, but also more secure.
+
+##### Limitations
+- The CA cert will be self-signed for this milestone.
+
+## Client CLI
+
+- Parse command line args into `client <job> <job_args>`.
+- Display usage and exit(1) on non-supported command or malformed args.
```

Comment:
How is signal termination handled here? If a process is terminate with a signal such as SIGKILL, the process does not have a normal exit code.